File: java/src/main/java/com/example/doubleup/model/Withdrawal.java
package com.example.doubleup.model;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import org.hibernate.annotations.GenericGenerator;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "withdrawal")
public class Withdrawal {
    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "player_id")
    private Player player;

    private OffsetDateTime createdAt;

    private Long amount;

    public Withdrawal() {
    }

    public Withdrawal(Player player, OffsetDateTime createdAt, Long amount) {
        this.player = player;
        this.createdAt = createdAt;
        this.amount = amount;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    public OffsetDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public Long getAmount() {
        return amount;
    }

    public void setAmount(Long amount) {
        this.amount = amount;
    }
}

File: go/model/withdrawal.go
package model

import (
    "time"
    "github.com/google/uuid"
)

type Withdrawal struct {
    ID        uuid.UUID
    PlayerID  uuid.UUID
    CreatedAt time.Time
    Amount    int64
}

func NewWithdrawal(playerID uuid.UUID, amount int64) *Withdrawal {
    return &Withdrawal{
        ID:        uuid.New(),       // Assuming you're using the github.com/google/uuid package for UUID generation
        PlayerID:  playerID,
        CreatedAt: time.Now().UTC(), // Using UTC time zone
        Amount:    amount,
    }
}
